src/core/tests/widgets/test_base.py

```diff
@@ -220,6 +312,7 @@ def test_repeat_set_app(self):
 
         # The app is still assigned to the original app
         self.assertEqual(self.widget.app, app)
+        self.assertEqual(list(app.widgets), [self.widget])
```

Comment:
We need to be a little careful with this as a test assertion. Lists are ordered, but we don't actually care about the order; just that the same contents exists. Checking using a `set()` would probably be a better option.

Reply:
In situations like this, `sorted` is better than `set`, because `set` would remove duplicates.

Reply:
If we've got duplicates in the IDs stored in the registry, we have a much bigger problem... :-)